x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -191,6 +249,17 @@ public LoggingAuditTrail(Settings settings, ClusterService clusterService, Threa
         this.events = parse(INCLUDE_EVENT_SETTINGS.get(settings), EXCLUDE_EVENT_SETTINGS.get(settings));
         this.includeRequestBody = INCLUDE_REQUEST_BODY.get(settings);
         this.threadContext = threadContext;
+        this.inferRealmNameFromUsername = (username) -> {
+            if (username == null) {
+                return null;
+            } else if (AnonymousUser.isAnonymousEnabled(settings) && AnonymousUser.isAnonymousUsername(username, settings)) {
+                return null;
+            } else if (ClientReservedRealm.isReserved(username, settings)) {
+                return ReservedRealm.NAME;
+            } else {
+                return NativeRealmSettings.NAME;
+            }
+        };
```

Comment:
This doesn't seem right.
Among other things, the native realm might be named something different, and also the reserved realm might be disabled.

If we really need to record a realm name for a user then I think we need to do it correctly.
Doing it in a possibly-not-correct way isn't helping anyone.

Reply:
🤦 It escaped me that you must name the native realm.

With the current code base it looks like we'll have to bend over backwards to pull the realm name here. I say it's not worth it.

I'm going to remove outputing the realm name in the current PR and raise an enhancement request for it; unless you feel otherwise, ofc.

Reply:
We've briefly discussed this and concluded it would be difficult to include the realm name in the audited events of APIs that operate on the native and reserved user stores. The user stores are decoupled from the realms and the APIs operate on the stores and are oblivious to the realm setup.
The audited event scheme is still of the form `{"user": {"name": .. }}` , instead of something like `{"username":...}`, to accommodate outputting the realm name in the future (eg from the response).